src/vs/workbench/contrib/positronNotebook/browser/selectionMachine.ts

```diff
@@ -396,12 +614,9 @@ export class SelectionStateMachine extends Disposable {
 		if (state.type === SelectionState.EditingSelection) {
 			return;
 		}
-		if (state.type === SelectionState.NoSelection) {
-			// Select first cell if selecting down and the last cell if selecting up.
-			const cellToSelect = cells.at(up ? -1 : 0);
-			if (cellToSelect) {
-				this.selectCell(cellToSelect, CellSelectionType.Normal);
-			}
+		// NoCells case: Cannot move selection when no cells exist
+		// Invariant ensures we're never in NoCells when cells exist
```

Comment:
IIUC `enforceInvariant` is only called when `cells` changes, not when the selection changes -- is this true?

Reply:
See https://github.com/posit-dev/positron/pull/9805#discussion_r2411419285